summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala

```diff
@@ -216,14 +243,23 @@ private[storm] case class StormTopologyBuilder(options: Map[String, Options], jo
     Topology.BoltId(getNodeName(node))
 
   private def shouldEmitKeyValues(node: StormNode): Boolean =
-    stormDag.dependantsOf(node).exists(_.isInstanceOf[SummerNode[_]])
+    stormDag.dependantsOf(node).exists {
+      case flatMapNode: FlatMapNode[Storm] => isGroupedLeftJoinNode(flatMapNode)
+      case summerNode: SummerNode[Storm] => true
+    }
 
   private def withLocalGrouping(node: FlatMapNode[Storm]): Boolean = {
     val usePreferLocalDependency = getOrElse(node, DEFAULT_FM_PREFER_LOCAL_DEPENDENCY)
     logger.info(s"[${getNodeName(node)}] usePreferLocalDependency: ${usePreferLocalDependency.get}")
     usePreferLocalDependency.get
   }
 
+  private def isGroupedLeftJoinNode(node: FlatMapNode[Storm]): Boolean =
+    node.members.last.isInstanceOf[LeftJoinedProducer[Storm, _, _, _]] &&
```

Comment:
@johnynek `.last` is here refers to first element in producers chain unfortunately (I added a comment on `.members` in previous PR).

I don't think we need tests on this edge cases because we don't support this edge cases yet.